app/src/main.cpp

```diff
@@ -495,6 +436,75 @@ namespace scitt
         .set_forwarding_required(ccf::endpoints::ForwardingRequired::Never)
         .install();
 
+      static constexpr auto get_entry_statement_path =
+        "/entries/{txid}/statement";
+      auto get_entry_statement = [this](
+                                   EndpointContext& ctx,
+                                   ccf::historical::StatePtr historical_state) {
+        SCITT_DEBUG("Get transaction historical state");
+        auto historical_tx = historical_state->store->create_read_only_tx();
+
+        auto entries = historical_tx.template ro<EntryTable>(ENTRY_TABLE);
+        auto entry = entries->get();
+        if (!entry.has_value())
+        {
+          throw BadRequestError(
+            errors::InvalidInput,
+            fmt::format(
+              "Transaction ID {} does not correspond to a submission.",
+              historical_state->transaction_id.to_str()));
+        }
+
+        auto proof = ccf::describe_merkle_proof_v1(*historical_state->receipt);
+        if (!proof.has_value())
+        {
+          throw InternalError("Failed to get Merkle proof");
+        }
+
+        auto signature = describe_cose_signature_v1(*historical_state->receipt);
+        if (!signature.has_value())
+        {
+          throw InternalError("Failed to get COSE signature");
+        }
+
+        // See
+        // https://datatracker.ietf.org/doc/draft-ietf-cose-merkle-tree-proofs/
+        // Page 11, vdp is the label for verifiable-proods in the unprotected
+        // header of the receipt
+        int64_t vdp = 396;
+        // -1 is the label for inclusion-proofs
+        auto inclusion_proof = ccf::cose::edit::pos::AtKey{-1};
+        ccf::cose::edit::desc::Value inclusion_desc{
+          inclusion_proof, vdp, *proof};
+
+        auto cose_receipt =
+          ccf::cose::edit::set_unprotected_header(*signature, inclusion_desc);
```

Comment:
This part looks the same as in 399-422. I suggest extracting this to a common method to reduce duplication.